pkg/api/controller.go

```diff
@@ -819,10 +829,37 @@ func (c *Controller) CreateUser(w http.ResponseWriter, r *http.Request, body Cre
 	}
 	ctx := r.Context()
 	c.LogAction(ctx, "create_user")
+	var userEmail string
+	invite := swag.BoolValue(body.InviteUser)
+	if invite {
+		normalizedEmail, err := regexp.Compile(body.Id)
+		e := normalizedEmail.String()
+		userEmail = strings.ToLower(e)
+		if err != nil {
+			return
+		}
+		u.Username = userEmail
+		u.Email = &userEmail
+
+		p, err := nanoid.New()
```

Comment:
Verify the following - I don't think we require generating a unique password and set it to the user.
Keep the password black, block the user ever to log in to the system until the password is reset.
We keep the encrypted password in the database, and empty is not an option that our hash function can return.
So, there is no need to set up a random password, as the user will not be able to log in to the system until the reset flow completes.

Reply:
Agree, removed the password creation and just left it empty.